sky/data/storage.py

```diff
@@ -437,15 +457,21 @@ def __init__(self,
         self.persistent = persistent
         self.mode = mode
         assert mode in StorageMode
+        if self.mode != StorageMode.CSYNC:
+            assert interval_seconds is None
+        self.interval_seconds = interval_seconds
         self.sync_on_reconstruction = sync_on_reconstruction
 
         # TODO(romilb, zhwu): This is a workaround to support storage deletion
         # for spot. Once sky storage supports forced management for external
         # buckets, this can be deprecated.
         self.force_delete = False
 
-        # Validate and correct inputs if necessary
-        self._validate_storage_spec(name)
+        if self.sync_on_reconstruction:
```

Comment:
Why only validate on `self.sync_on_reconstruction`?

Reply:
This and elif name are old piece of codes. Removed them. Thanks for catching.